workspace.spec

```diff
@@ -572,6 +572,466 @@ module Workspace {
+
+		The 'resource_type' field should be filled using values from the DataCite resourceTypeGeneral field:
+
+		https://support.datacite.org/docs/datacite-metadata-schema-v44-mandatory-properties#10a-resourcetypegeneral
+
+		Currently the KBase workspace only supports credit metadata for objects of type 'dataset'; anything else will return an error.
+
+		Required fields are:
+		- identifier
+		- resource_type
+		- versioning information: if the resource does not have an explicit version number, one or more dates should be supplied: ideally the date of resource publication and the last update (if applicable).
+		- contributors (one or more required)
+		- titles (one or more required)
+
+		comments - freeform text providing extra information about this credit metadata.
+			Examples:
+				- Credit metadata generated automatically from DOI:10.13039/100000015
+
+		identifier - persistent unique identifier for the resource (i.e. the source data for this workspace object).
+			Should be in the format <database name>:<identifier within database>
+			Examples:
+				- RefSeq:GCF_004214875.1
+				- GenBank:CP035949.1
+				- img.taxon:648028003
+
+		license (optional) - usage license for the resource. May be a text string or an URL. Abbreviations should be spelled out where possible (e.g. "Creative Commons 4.0" instead of "CC-BY-4.0").
+			Examples:
+				- Creative Commons 4.0
+				- MIT
+				- https://jgi.doe.gov/user-programs/pmo-overview/policies/
+
+		resource_type - the broad type of the source data for this workspace object.
+			Valid 'resource_type' values:
+				- dataset
+
+		schema_version - version of the credit metadata schema used.
+			Example:
+				- 1.1.0
+
+		version (optional if dates are provided) - the version of the resource. This must be an absolute version, not a relative version like 'latest'.
+			Examples:
+				- 5
+				- 1.2.1
+				- 20220405
+
+		contributors - a list of people and/or organizations who contributed to the resource.
+
+		dates (optional if version is provided) - a list of relevant lifecycle events for the resource.
+
+		funding (optional) - funding sources for the resource.
+
+		related_identifiers (optional) - other resolvable persistent unique IDs related to the resource.
+
+		titles - one or more titles for the resource.
+
+	*/
+	typedef structure {
+		string comments;
+		string identifier;
+		string license;
+		string resource_type;
+		string schema_version;
+		string version;
+		list<Contributor> contributors;
+		list<EventDate> dates;
+		list<FundingReference> funding;
+		list<PermanentID> related_identifiers;
+		list<Title> titles;
+	} CreditMetadata;
+
+	/* CreditMetadataEntry
+
+		Container for an instance of credit metadata; returned by the workspace when credit metadata is requested.
+
+		All fields will be populated.
+
+		credit_metadata - the credit metadata itself
+
+		credit_metadata_schema_version - version of the credit metadata schema used
```

Comment:
Why is the schema version in `CreditMetadata` and `CreditMetadataEntry`?

Reply:
I want `credit_metadata_schema_version` to be submitted by the user when they add the credit metadata, i.e.

```json
{
    "credit_metadata_schema_version": "1.1.0",
    "credit_metadata": { ... }
}
```

When a submission is made, `saved_by` and `timestamp` fields will be added automatically, and the saved data structure will look like this:

```json
{
    "credit_metadata_schema_version": "1.1.0",
    "saved_by": "user_name_here",
    "timestamp": 1234567890,
    "credit_metadata": { ... }
}
```

This structure is what should be returned when users request credit metadata for a WS object.

I think I misinterpreted something you said and that I need to reinstate the spec for this data structure:
```json
{
    "credit_metadata_schema_version": "1.1.0",
    "credit_metadata": { ... }
}
```
and undo the addition of the `credit_metadata_schema_version` from `CreditMetadata`.

Reply:
Looks like the schema version was removed from `CreditMetadata` so the OQ is moot.

> What would be your proposal for a function for adding credit metadata to an object, and where would it go -- in this workspace.spec file or elsewhere?

I would use the extant `administer()` method.